Sources/StreamChatUI/Utils/TextViewMentionedUserDetector.swift

```diff
@@ -0,0 +1,18 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class TextViewMentionedUserDetector {
+    open func handleMentionedUserInteraction(on textView: UITextView, in characterRange: NSRange, _ mentionedUsers: Set<ChatUser>?, onTap: ((ChatUser?) -> Void)?) {
+        guard let text = textView.text,
+              let range = Range(characterRange, in: text)
+        else { return }
+        
+        let string = String(text[range].replacingOccurrences(of: "@", with: ""))
+        let user = mentionedUsers?.first(where: { $0.name == string })
+        onTap?(user)
+    }
+}
```

Comment:
-  `onTap: ((ChatUser?) -> Void)?` should be a reference property, or it could also be a delegate. But closure are more friendly on SDKs so that we don't break the API easily.
- `mentionedUsers: Set<ChatUser>?` should not be nullable. If there are no mentionedUsers, then don't call this method. 

Here is a suggestion of improvement:
```suggestion
open class TextViewMentionedUsersHandler {
    var onMentionedUserTap: ((ChatUser?) -> Void)? // This can also be a delegate

    open func handleInteraction(
        on textView: UITextView,
        in characterRange: NSRange,
        withMentionedUsers mentionedUsers: Set<ChatUser> // This should not be nil, if there are no mentioned users, don't call this method
    ) {
        guard let text = textView.text,
              let range = Range(characterRange, in: text)
        else { return }

        let string = String(text[range].replacingOccurrences(of: "@", with: ""))
        let user = mentionedUsers?.first(where: { $0.name == string })
        onMentionedUserTap?(user)
    }
}
```

Then, in the `ComposerVC` we simply handle the closure of `TextViewMentionedUsersHandler`, and nothing else. In the `ChatMessageListVC` we handle the closure and forward it to the `ChatMessageContentViewDelegate` just to make it easier on the message list. With this, we can handle interaction in both components, and on the message list, we follow the current pattern to handle message content view events.

Let me know if you have any questions.